### PR TITLE
fix(datasource/galaxy): support multiple results

### PR DESCRIPTION
**Pull request by rarkins**
## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/datasource/galaxy/index.ts [hunk 1]
```diff
@@ -50,13 +50,19 @@ export class GalaxyDatasource extends Datasource {
...
     }
 
-    // istanbul ignore if
     if (body.results.length > 1) {
-      logger.warn(
-        { dependency: packageName },
-        `Received multiple results from ${galaxyAPIUrl}`,
-      );
-      return null;
+      const githubUsernames = body.results.map((result) => result.github_user);
```
Comment (viceice) [inline lib/modules/datasource/galaxy/index.ts:54]: I would filter here and warn if the filtered result is empty. that way we only have to iterate once.

File: lib/modules/datasource/galaxy/index.ts [hunk 2]
```diff
@@ -50,13 +50,19 @@ export class GalaxyDatasource extends Datasource {
       throw this.handleGenericErrors(err);
     }
```
Comment (secustor) [inline lib/modules/datasource/galaxy/index.ts:52]:
```suggestion
     body.results = body.results.filter((result) => result.github_user === username)
```

File: lib/modules/datasource/galaxy/index.ts [hunk 3]
```diff
@@ -50,13 +50,19 @@ export class GalaxyDatasource extends Datasource {
...
+        body.results = body.results.filter(
+          (result) => result.github_user === userName,
+        );
+      } else {
+        logger.warn(
+          { dependency: packageName, githubUsernames },
+          `Multiple results from galaxy for ${packageName}, none match`,
+        );
+        return null;
+      }
```
Comment (secustor) [inline lib/modules/datasource/galaxy/index.ts:65]:
```suggestion
      logger.warn(
        { dependency: packageName },
        `Received multiple results from ${galaxyAPIUrl}`,
      );
      return null;
```

**Review by secustor — DISMISSED** [review]

File: lib/modules/datasource/galaxy/index.ts [hunk 4]
```diff
@@ -49,8 +49,7 @@ export class GalaxyDatasource extends Datasource {
     } catch (err) {
       this.handleGenericErrors(err);
     }
-
-    // istanbul ignore if
+     body.results = body.results.filter((result) => result.github_user === username)
```
Comment (rarkins) [inline lib/modules/datasource/galaxy/index.ts:52]:
```suggestion
    body.results = body.results.filter((result) => result.github_user === username)
```

**Review by viceice — APPROVED** [review]

**Pull request merged**